### PR TITLE
Move userdata name into MCS status

### PR DESCRIPTION
**Pull request by enxebre**
At the moment the MCS userdata name is assumed by the machineDeployment as an arbitrary convention.
This solidify that contract through the API satatus providing the ability to change this name convention in the future.

### CODE REVIEW

File: hypershift-operator/controllers/machineconfigserver/manifests.go [hunk 1]
```diff
@@ -55,11 +56,11 @@ func MachineConfigServerIgnitionRoute(machineConfigServerNamespace, machineConfi
 	}
 }
 
-func MachineConfigServerUserDataSecret(mco *hyperv1.MachineConfigServer) *corev1.Secret {
+func MachineConfigServerUserDataSecret(mcs *hyperv1.MachineConfigServer) *corev1.Secret {
 	return &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: mco.GetNamespace(),
-			Name:      fmt.Sprintf("user-data-%s", mco.GetName()),
```
Comment (ironcladlou) [inline hypershift-operator/controllers/machineconfigserver/manifests.go:62]: Any specific reason for changing the secret name? The new name no longer suggests much about what the secret might be used for (not that it matters much, unless there comes a time we need two secrets for the MCS...)

Comment (enxebre) [reply hypershift-operator/controllers/machineconfigserver/manifests.go:62]: we might need two at some point, but for such a case we'd probably want to append the version or similar. I'll put it back as you mentioned to clarify its purpose

**Pull request merged**